### ISSUE TITLE
Account settings | No UI validations or restrictions if the User tries to change password

### ISSUE THREAD

**Issue by ddavid9755** [mattermost/mattermost-server#8506]
Per Mattermost guidelines, GitHub issues are for bug reports: http://www.mattermost.org/filing-issues/

For troubleshooting see: http://forum.mattermost.org/.
For feature proposals see: http://www.mattermost.org/feature-requests/

If you've found a bug--something appears unintentional--please report using the following format: 

#### Summary
If the User want to change his password, there are no UI validations about the new password. The user can add for example 'spaces' to the new password field. No need for numbers, characters etc.

#### Steps to reproduce
1. Log in to https://oss.mattermost.com/joomla/channels/town-square
2. Click to the hamburger icon near your profile name
3. Select the 'Account settings' option
4. Select 'security' option
5. Click to 'password'
6. Type in your current password
7. Type in a new password, for example, 5 space-s
8. Click save button

Note: If you try to type is: asdasd, it's the same, you can save it without any notification.

#### Expected behavior
In the registration process there are several restrictions about creating password. If the user is trying to change his password, the method should be the same.

#### Observed behavior (that appears unintentional) 
Please see the uploaded video below:
https://screencast.com/t/BDZQRvkdilMe

Note:
In the video you can see, there are several errors shown in the console.

#### Possible fixes
If you can, link to the line of code that might be responsible for the problem

**Comment by amyblais** [comment]
Hey @ddavid9755,

Thank you for submitting this Issue!

Could you help clarify the sentence in the "Expected behavior" (`In the registration process there are several restrictions about creating password. If the user is trying to change his password, the method should be the same`) - what restrictions are you seeing when creating an account vs. when updating the password?

FYI, we have [password restrictions in EE](https://docs.mattermost.com/administration/config-settings.html#password), and here's previous discussion on why: https://github.com/mattermost/mattermost-server/issues/5935.

**Comment by ddavid9755** [comment]
Hi amyblais,
Sorry, this Item should be Invalid because of the following:
- I forgot to check the registration process again
- I remembered that, there is a restriction in the registration process to free users (I checked it in Chrome, Firefox)

I read the linked pages. 
After I did a quick smoke testing, I missed to read the configuration page. This Item would help to improve security standards, but after I read the links, I realised how the whole works.

Thank you for your comment.
